pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/datasource/DataSourceMetadata.java

```diff
@@ -70,6 +70,11 @@ default boolean isSingleValue() {
    */
   int getMaxNumValuesPerMVEntry();
 
+  /**
+   * Returns whether forward index is disabled for this column or not, returns 'true' if disabled
+   */
+  boolean forwardIndexDisabled();
```

Comment:
I don't think we need this in the metadata. Instead, we can make `DateSource.getForwardIndex()` nullable, and perform proper null checks when it is used. This is consistent with how we handle other indexes.

Reply:
i removed this as it isn't used at the moment